src/process/process.rs

```diff
@@ -0,0 +1,52 @@
+#![allow(dead_code)] // FIXME
+
+use anyhow::{anyhow, Result};
+
+use super::cli::Process;
+use crate::{
+    cli::CliConfig,
+    core::events::file::FileEventsFactory,
+    module::{get_modules, Group},
+};
+
+pub(crate) struct PostProcess {
+    modules: Group,
+    cli: CliConfig,
+}
+
+impl PostProcess {
+    pub(crate) fn new(cli: CliConfig) -> Result<Self> {
+        let config = cli
+            .subcommand
+            .as_any()
+            .downcast_ref::<Process>()
+            .ok_or_else(|| anyhow!("wrong subcommand"))?;
+
+        let factory = FileEventsFactory::new(&config.file);
+        let mut modules = get_modules(Box::new(factory))?;
+
+        modules.start_factory()?;
+
+        // Read all events from the factory.
+        let mut events = Vec::new();
+        while let Some(event) = modules.factory.next_event(None)? {
+            events.push(event);
```

Comment:
I don't think putting all the events in memory is a good idea. It's likely that, for many usecases the size of this will be very big.